src/node/tendermint_internals.mli

```diff
@@ -0,0 +1,94 @@
+open Crypto
+
+type height = int64 [@@deriving yojson]
+
+type round = int [@@deriving yojson]
+(** At a specific (chain) height, Tendermint's consensus algorithm may run several rounds.*)
+
+(** Tendermint sometimes decides on a `Nil` value. *)
+type value =
+  | Block of Protocol.Block.t
+  | Nil
+[@@deriving yojson]
+
+val repr_of_value : value -> value
```

Comment:
This function does not make sense. As it is, it is pointless.

Reply:
Yes, as it is it's pointless. However, it shouldn't be. Tendermint is meant to use `repr_of_value` and not the actual value, as a way to minimize what's sent on the network. We used `repr_of_value` in the code to remind ourselves of that so we'll implement it in the future.

Reply:
Yeah, Ulrik beat me to it, but what I think you want is an abstract type `repr`. Then `repr_of_value` makes total sense. Under the hood, you can implement it as the identity function, and later you can change it to be more efficient without changing the signature at all.